File: Data Structures & Algorithms/binary-tree-diameter/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int res = 0;
    public int diameterOfBinaryTree(TreeNode root) {
        int ans = dfs(root);
        return this.res;
        
    }

    public int dfs(TreeNode curr){
        if(curr==null){
            return 0;
        }

        int left = dfs(curr.left);
        int right = dfs(curr.right);
        this.res = Math.max(left+right, res);
        return Math.max(left,right)+1;
    }
}

File: Data Structures & Algorithms/valid-parenthesis-string/submission-1.py
class Solution:
    def checkValidString(self, s: str) -> bool:
        left = 0
        right = 0

        for ch in s:
            if ch=='(':
                left+=1
                right+=1
            elif ch == ')':
                left-=1
                right-=1
            else:
                left-=1
                right+=1
            if right < 0:
                return False
            left = max(left, 0)
        return left == 0